community/modules/compute/schedmd-slurm-gcp-v5-partition/main.tf

```diff
@@ -69,6 +69,15 @@ locals {
   # Also, slurm imposed a lot of restrictions to this name, so we format it to an acceptable string
   tmp_cluster_name   = substr(replace(lower(var.deployment_name), "/^[^a-z]*|[^a-z0-9]/", ""), 0, 8)
   slurm_cluster_name = var.slurm_cluster_name != null ? var.slurm_cluster_name : local.tmp_cluster_name
+
+  uses_zone_policies = length(var.zone_policy_allow) + length(var.zone_policy_deny) > 0
+  excluded_zones     = var.zone == null ? [] : [for z in data.google_compute_zones.available.names : z if z != var.zone]
+  zone_policy_deny   = local.uses_zone_policies ? var.zone_policy_deny : local.excluded_zones
+}
+
+data "google_compute_zones" "available" {
```

Comment:
What happens if a new zone become available while this cluster is running? will bulk api start scheduling work there?

Reply:
That's a good question. Since we are creating a single list based on what's known at the time, if a new zone is created in the region, my assumption is that it would be set to allow by default (as all zones are) and could lead to a risk of egress fees.

I don't know of a good way to ensure all created zones are denied. @tpdownes, do you have any thoughts on this?

This would only happen if the zone was entirely unknown at deploy time, as all zones, up or down, are captured by this terraform data structure. I'll add some more documentation to clarify this risk. I still believe having this option is worth it, since this risk is rare and it greatly reduces the overhead for a user to specify a single zone.

Reply:
The question really is, what does zone_policy_allow do? can you use that instead of zone_policy_deny? 

Reply:
From my research and experience, zone_policy_allow doesn't do much more than make the default explicit. 

In this case, it wouldn't change anything functionally, but it would make sense to explicitly define it. I'll follow up with that after the dependent PR is merged. 

The only clear way I've found of guaranteeing a zone is to use [instances.bulkInsert] rather than [regionInstances.bulkInsert] when `zone` is specified. This would require a change to the logic of the resume script.

[instances.bulkInsert]: https://cloud.google.com/compute/docs/reference/rest/v1/instances/bulkInsert
[regionInstances.bulkInsert]: https://cloud.google.com/compute/docs/reference/rest/v1/regionInstances/bulkInsert